treerec/tests/conftest.py

```diff
@@ -135,11 +135,12 @@ def individual_output(self):
             fields = line.split()
             assert fields[0].startswith("#Individual:")
             store = fields[0][len("#Individual:"):]
-            assert store in ("remember", "retain", "output")
+            assert store in ("remember", "retain", "retain_if_unary", "output")
             pedigree_id = int(fields[1])
-            if pedigree_id in slim and store == "retain":
-                # We have a duplicate: "remember"ed takes priority
-                continue
+            if pedigree_id in slim:
+                # We have a duplicate:
+                if slim[pedigree_id].type == "remember":
+                    continue  # "remember"ed takes priority: do not overwrite
```

Comment:
not getting the whole picture here, but do we also need a "retain takes priority over retain if unary" here?

Reply:
oh, I see, we only check below if this starts with retain, so it doesn't matter? still, it'd be nice to get the right thing in here.

Reply:
We never have both `retain_if_unary` and `retain` in the same simulation, so it's never an issue.

Reply:
ah, well, perhaps we should have them both in the same simulation?

Reply:
Since we only specify "retainCoalescentOnly" for the entire TS recording, we have no way of doing this. But I can change the logic in case we ever do implement something similar? I can't see how it would work though!

Reply:
Sorry, maybe I'm just confused, but can't we do `retainCoalescentOnly=T` up top, and then insert some calls to `treeSeqRememberIndividuals(..., permanent=F)` in the recipe?

Reply:
Yes we can do that. If so, we need to mark all retained individuals as "retain". If, at the top, we use `retainCoalescentOnly=F`, we save details of all retained individuals under `retain_if_unary`.

Reply:
Note that "retain" and "retain_if_unary" are only convenience flags in the text-save of the individuals, for testing. And also note that any individuals we zap with `treeSeqRememberIndividuals(..., permanent=T)` are marked as "remember" which always overrides either "retain" marking.

Reply:
I think I'm being slow here, but: we can do `retainCoalescentOnly=T`, and then only mark SOME of the individuals to retain; this will produce a mix of `retain` and `retain_if_unary`. Ah, I see - if `retainCoalescentOnly=T`, then should `retain` have no effect? (Still, we could be checking nothing weird happens.)

Reply:
I think there's some confusion here. `retainCoalescentOnly=T` only applies during `treeSeqInitialize()`. So either it's on for the whole simulation, or off for the whole simulation.

If "on", then any node permanently remembered is output with a "remember" tag, and and node temporarily retained is output with a "retain" flag.

If "off", then any node permanently remembered is output with a "remember" tag, and and node temporarily retained is output with a "retain_if_unary" flag.

Reply:
ack, sorry, I think I need some more coffee now.